pkg/rtc/transport.go

```diff
@@ -337,12 +352,37 @@ func (t *PCTransport) Close() {
 
 func (t *PCTransport) SetRemoteDescription(sd webrtc.SessionDescription) error {
 	t.lock.Lock()
-	defer t.lock.Unlock()
+
+	var (
+		iceCredential   string
+		offerRestartICE bool
+	)
+	if sd.Type == webrtc.SDPTypeOffer {
+		var err error
+		iceCredential, offerRestartICE, err = t.isRemoteOfferRestartICE(sd)
+		if err != nil {
+			t.Logger().Errorw("check remote offer restart ice failed", err)
```

Comment:
should we still set the remote description anyways? thinking if there's a scenario where parsing failed, but the offer is actually valid.

Reply:
pion do the parse in `SetRemoteDescription` too, it will fail if parse failed.